src/lib.rs

```diff
@@ -3718,7 +3718,7 @@ impl Build {
         if let Some(val) = self.env_cache.read().unwrap().get(v).cloned() {
             return val;
         }
-        if self.emit_rerun_if_env_changed && !provided_by_cargo(v) {
+        if self.emit_rerun_if_env_changed && !provided_by_cargo(v) && v != "PATH" {
```

Comment:
Someone looking at this code in the future will be very confused about why PATH got special-cased here. Would have been good to add a comment. :)

Reply:
Yeah I could add one there

Reply:
Opened #1247